package/MDAnalysis/lib/pkdtree.py

```diff
@@ -176,7 +178,8 @@ def search(self, centers, radius):
         centers = np.asarray(centers)
         if centers.shape == (self.dim, ):
             centers = centers.reshape((1, self.dim))
-
+        if self.cutoff is None:
```

Comment:
`self.cutoff` can be `None` is `self.pbc` is `False`. It is only if `self.pbc == True` that we assume we have a value for the cutoff.

Reply:
If i use 
if self.pbc and (self.cutoff is None) :
raise Valuerror("Provide cutoff")

mypy still raises this error 
Unsupported operand types for > ("float" and "None")
Left operand is of type "Optional[float]"

Reply:
Can you push the code with the conditional moved? This way I can comment on the code you have rather than the code I imagine you may have. Also, it will let us see if the tests, other than mypy, pass.

Reply:
This remains outstanding. It looks like it is the latest obstacle in terms of type annotations.

Reply:
I did not realise you did push your latest change. Here is a possible solution. I did not test it, but it might work:

```python
if self.pbc:
    if self.cutoff is None:
        raise ValueError("Cutoff needs to be provided when working with PBC.")
    cutoff: float = self.cutoff
    # From here and until the end if the if self.pbc block, use cutoff instead of self.cutoff
```

I haven't tested that solution, but I expect that mypy will figure out that `cutoff` can only be a float.

Reply:
I tried that but it still raises the same error

Reply:
Done